firebase/src/core/data/gateways/firestore-collection-name.ts

```diff
@@ -0,0 +1,22 @@
+/** Typed firestore collections. */
+export type FirestoreCollectionName =
+  | PublicCollectionName
+  | PublicSubCollectionName
+  | UserCollectionName
+  | UserSubCollectionName;
+
+/** Public collections available to all authenticated users. */
+type PublicCollectionName = "public_collections" | "public_collection_categories" | "public_memos";
```

Comment:
is the `public` suffix necessary since they'll be placed in the root and the private ones under `users/`? It's seems intuitive IMO

Reply:
Yeah, we should be able to differ from collection groups that belong to users versus public ones. If we don't have this distinction, when running collection groups on `collections`, we would not only be reading all user's collections, but on public one's as well.

Reply:
I can't remember why I did this, as I've had a hiatus of about 6 weeks on this project, that's completely on me. My bad. I will rollback to something that still make sense.

I think it was related to a possible future marketplace feature, but it's so far-fetched that it makes no sense to do something like that right now